### PR TITLE
[SPARK-43342][K8S] Revert SPARK-39006 Show a directional error message for executor PVC dynamic allocation failure

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This reverts commit b065c945fe27dd5869b39bfeaad8e2b23a8835b5.

### Why are the changes needed?

To remove the regression from SPARK-39006.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs.

Closes #41057

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @dcoliversun This will be merged with your authorship.

**Comment by dongjoon-hyun** [comment]
After merging this clean-revert, you may want to make a new and proper test PR for https://github.com/apache/spark/pull/41057/files#r1186469992 if you want.

**Comment by dcoliversun** [comment]
@dongjoon-hyun Thanks for your PR. I'm +1

**Comment by dongjoon-hyun** [comment]
Thank you, @dcoliversun .

**Comment by dongjoon-hyun** [comment]
Could you review this PR, @viirya ?

**Comment by viirya** [comment]
May I ask why this is reverted?

**Comment by dongjoon-hyun** [comment]
Sorry for missing context.

SPARK-43329 and SPARK-43342 are filed as a regression at Apache Spark 3.4.0.

I linked all the reported issues to the original JIRA which causes this regression.
- https://issues.apache.org/jira/browse/SPARK-39006 Show a directional error message for PVC Dynamic Allocation Failure

Screenshot 2023-05-06 at 5 13 51 PM



And, this was the reverting PR from the author of SPARK-39006.
- https://github.com/apache/spark/pull/41057


In short, while adding a log message, we missed a corner case which one NFS-backed PVC is mounted by multiple executors. So, we decide to revert SPARK-39006 because it doesn't give actual benefit (except logging).

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Thank you @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
Thank you!

**Comment by dongjoon-hyun** [comment]
Merged to master/3.4.
